app/domain/ca/ssh/verify.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
```

Comment:
This is currently true... but this is somewhat preparatory for what's coming next. This is draft code, but there is going to be additional verification of permissions, as seen in the WIP branch here: https://github.com/cyberark/conjur/blob/4772a9757e333b0e08d6e5864901d8b9c05cb974/app/domain/ca/x509/verify.rb#L54-L87

Would you move all of this to the controller? This could be worth a call, but I start to get a little hazy on where all of the code for pulling things from the database should live...

Reply:
> where all of the code for pulling things from the database should live...

So, yes.  As you should.  It should live in repositories, as in the Repository pattern -- which you can query and it returns proper domain objects for you.  The problem is, this is conflict with rails and active record.  I've used it a few places anyway, I think it's fine but it's debatable.  I mean, if you're writing good clear code that won't confuse anyone, and add clarifying comments, I feel alright about it.

You two choices:

1. Screw it and create a Repo object 
2. Add methods to the rails model that return your domain objects

The important thing is to isolate the database interaction to one place so that everywhere else in your code can just use value objects without fear, and all the application code will be nice.